src/ucp/wireup/address.c

```diff
@@ -1515,9 +1539,10 @@ ucp_address_length(ucp_worker_h worker, const ucp_ep_config_key_t *key,
     }
 
     /* Calculate the required ucp address length */
-    *size_p = ucp_address_packed_size(worker, devices, num_devices, pack_flags,
+    size    = ucp_address_packed_size(worker, devices, num_devices, pack_flags,
                                       addr_version);
-    status  = UCS_OK;
+    *size_p = size;
+    status  = (size > -1 ? UCS_OK : UCS_ERR_UNSUPPORTED);
```

Comment:
1. assign *size_p only if successful
2. preserve the original error code

Reply:
fixed